File: src/main/java/liqp/filters/Upcase.java
package liqp.filters;

import liqp.nodes.RenderContext;

class Upcase extends Filter {

    /*
     * upcase(input)
     *
     * convert a input string to UPCASE
     */
    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        return super.asString(value).toUpperCase();
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/Upcase.kt
package liqp.filter

import liqp.context.LContext
import liqp.params.FilterParams

class Upcase : LFilter("upcase", "upper") {

  /**
   * upcase(input)
   *
   * convert a input string to UPCASE
   */
  override fun onFilterAction(context: LContext, value: Any?, params: FilterParams): Any? {
    return context.asString(value)?.toUpperCase()
  }
}
